addons/point_of_sale/models/product.py

```diff
@@ -119,18 +119,28 @@ def _load_pos_data(self, data):
             products = config_id.with_context({**self.env.context, 'display_default_code': False}).get_limited_products_loading(fields)
         else:
             domain = self._load_pos_data_domain(data)
```

Comment:
```suggestion
            domain = expression.OR(self._load_pos_data_domain(data) + [('id', 'in', list({line['product_id'] for line in data['pos.order.line']['data']}))])
```
would something like this work?

Reply:
Hello @vlst-odoo 
It's not complete. The missing product must also be loaded with limited product loading. Additionally, it needs to load the archived products that were used in POS order lines. Therefore, I think the best way is to load them separately.